server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -161,24 +161,32 @@ public Task register(String type, String action, TaskAwareRequest request, boole
             Task previousTask = tasks.put(task.getId(), task);
             assert previousTask == null;
             if (traceRequest) {
-                startTrace(threadContext, task);
+                maybeStartTrace(threadContext, task);
             }
         }
         return task;
     }
 
+    // Start a new trace span if a parent trace context already exists.
+    // For REST actions this will be the case, otherwise Tracer#startTrace can be used.
```

Comment:
Could you make this into proper Javadoc so it's visible on mouseover in the IDE?

Reply:
done